### DISCUSSION TITLE
Compatibility with Quarto

### DISCUSSION THREAD

**Discussion by jhconning** [microsoft/vscode-jupyter#9630] [General] [answered]
The Quarto notebook project builds part of its documentation around how to use vs-code with the jupyter extension to create beautifully rendered HTML and pdf documents   https://quarto.org/docs/get-started/hello/vscode.html

However, I've noticed a problem with using Quarto syntax that results in a katex parsing error in a markdown cell when using the juptyer extension. The issue does not arise using JupyterLab.   

Quarto allows one to label and cross-reference equations, as described here: 
https://quarto.org/docs/get-started/authoring/vscode.html#cross-references

In the example there, a `{#eq-stddev}` label is added immediately after the `$$` delimiter (does not work if the label is on the next line).  The rendered quarto document then produces a numbered equation and the `@eq-stddev` label can be used elsewhere in the document to create clickable cross-reference links.

```markdown
## Equation 
THis is the standard deviation formula:
$$
\sqrt{\frac{1}{N-1} \sum_{i=1}^N (x_i - \overline{x})^2}
$$ {#eq-stddev}
```

The problem is that the juptyer extension in vs-code chokes on this syntax, and presents a Katex parser error instead of the rendered cell:
<img width="701" alt="image" src="https://user-images.githubusercontent.com/5048607/162324121-5723c7a1-044a-4607-9e2a-957bc85ccc96.png">

This problem does not arise in Jupyterlab, where the same is rendered like this:
<img width="401" alt="image" src="https://user-images.githubusercontent.com/5048607/162324008-c1d3a2a8-adf5-4e93-828a-69fbf3e27254.png">

Would it be possible to change the parsing to allow this syntax?    One could then develop notebooks using the jupyter extension and also have the option of using the powerful extra directives and syntax from quarto to produce beautiful websites, books, blog posts, and other documents.

**Comment by DonJayamanne** [comment, marked as answer]
Thanks, looks like a bug in the way we parse these latex expressions.
I've filed a bug for this in VS Code, & you can track that issue here https://github.com/microsoft/vscode/issues/147043
